geniza/corpus/models.py

```diff
@@ -851,6 +854,15 @@ def has_transcription(self):
     has_transcription.short_description = "Transcription"
     has_transcription.boolean = True
 
+    def has_translation(self):
+        """Does document have a translation?"""
+        return any(
+            [Footnote.TRANSLATION in note.doc_relation for note in self.footnotes.all()]
+        )
+
+    has_translation.short_description = "Translation"
+    has_translation.boolean = True
+
```

Comment:
looks fine; the short description and boolean are for django admin, so if we're not using it there we don't need them

needs a unit test — parallel to the has transcription one I imagine

(now I'm wondering how much effort to generalize `has_x` for all our types; maybe something for the next time we add one)

Reply:
> looks fine; the short description and boolean are for django admin, so if we're not using it there we don't need them

I see, gotcha

> needs a unit test — parallel to the has transcription one I imagine

How about this:
```python
def test_has_translation(self, document, source):
        # doc with no footnotes doesn't have translation
        assert not document.has_translation()

        # doc with empty footnote doesn't have translation
        fn = Footnote.objects.create(content_object=document, source=source)
        assert not document.has_translation()

        # doc with translation footnote does have a translation
        fn.doc_relation = [Footnote.TRANSLATION]
        fn.save()
        assert document.has_translation()
```

> (now I'm wondering how much effort to generalize `has_x` for all our types; maybe something for the next time we add one)

I agree that this is a good formulation and yeah let's keep it in mind if we end up expanding this/related fields.